cloudflare-speedtest/cloudflare-speedtest.sh

```
@@ -0,0 +1,66 @@
+timestamp=$(jq -r '.timestamp_utc // ""' "${file}")
+
+echo "$(date -Iseconds) speedtest results"
+
+echo "$(date -Iseconds) download = ${download} Mbps"
+echo "$(date -Iseconds) upload =  ${upload} Mbps"
+echo "$(date -Iseconds) ping =  ${ping} ms"
+echo "$(date -Iseconds) jitter = ${jitter} ms"
+
+echo "$(date -Iseconds) sending results to ${MQTT_HOST} as clientID ${MQTT_ID} with options ${MQTT_OPTIONS} using user ${MQTT_USER}"
+
+/usr/bin/mosquitto_pub -h ${MQTT_HOST} -p ${MQTT_PORT} -r -i ${MQTT_ID} ${MQTT_OPTIONS} -u ${MQTT_USER} -P ${MQTT_PASS} -t ${MQTT_TOPIC}/cloudflare-speedtest-download/config -m "{\"name\":\"Cloudflare Speedtest - Download\", \"state_topic\":\"${MQTT_TOPIC}/test\", \"value_template\":\"{{ value_json.download }}\", \"json_attributes_topic\": \"${MQTT_TOPIC}/test\", \"unit_of_measurement\":\"Mbps\", \"icon\":\"mdi:speedometer\", \"unique_id\":\"cloudflare_speedtest_download\"}"
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)